### PR TITLE
Assign a single meta at puzzle creation time

### PR DESCRIPTION
**Pull request by maximized**

### CODE REVIEW

**Comment by maximized** [comment]
image

image

image

**Comment by maximized** [comment]
The dropdown for meta options are also subject to https://github.com/cardinalitypuzzles/cardboard/issues/672

File: hunts/src/AddPuzzleModal.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 import Form from "react-bootstrap/Form";
 import { addPuzzle } from "./puzzlesSlice";
 import { hideModal } from "./modalSlice";
+import { selectHuntTags } from "./huntSlice";
 
 function AddPuzzleModal({ huntId }) {
+  const allTags = useSelector(selectHuntTags);
   const [name, setName] = React.useState("");
   const [url, setUrl] = React.useState("");
+  const [assignedMeta, setAssignedMeta] = React.useState("none");
```
Comment (rgossiaux) [inline hunts/src/AddPuzzleModal.js:14]: What's wrong with using a falsy value for this?

File: chrome_extension/cardboard/manifest.json [hunk 2]
```diff
@@ -16,8 +16,8 @@
 	"cookies"
   ],
   "host_permissions": [
-	"0.0.0.0:*",
-	"http://localhost:*/*",
+	"http://*:*/*",
```
Comment (rgossiaux) [inline chrome_extension/cardboard/manifest.json:19]: This feels kind of sketchy, was this just to include 127.0.0.1 or was there more? I guess it's ok as a temporary hack if we need to

File: hunts/src/AddPuzzleModal.js [hunk 3]
```diff
@@ -53,6 +58,25 @@ function AddPuzzleModal({ huntId }) {
               onChange={(e) => setUrl(e.target.value)}
             />
           </Form.Group>
+          <Form.Group controlId="addPuzzle.meta">
+            <Form.Label>Assigned Meta</Form.Label>
+            <Form.Control
+              as="select"
+              value={assignedMeta}
```
Comment (akirabaruah) [inline hunts/src/AddPuzzleModal.js:65]: If you use a falsy value like `null` (@rgossiaux's comment above), you can probably change this to `value={assignedMeta ?? "none")`.

File: chrome_extension/cardboard/manifest.json (hunk 2 above)
Comment (akirabaruah) [reply chrome_extension/cardboard/manifest.json:19]: Also, would this potentially get flagged during the Chrome Extension store review?

File: api/views.py [hunk 4]
```diff
@@ -375,6 +375,16 @@ def create(self, request, **kwargs):
...
 
+            if (
+                "assigned_meta" in request.data
+                and request.data["assigned_meta"]
+                and request.data["assigned_meta"] != "none"
+            ):
+                meta = get_object_or_404(
+                    Puzzle, name=request.data["assigned_meta"], hunt=puzzle.hunt
+                )
+                puzzle.metas.add(meta)
```
Comment (akirabaruah) [inline api/views.py:386]: Should we add a unit test for this? (Can be in a separate PR if we want to expedite this one)

File: hunts/src/AddPuzzleModal.js (hunk 1 above)
Comment (maximized) [reply hunts/src/AddPuzzleModal.js:14]: Nothing. Should I use the falsy value for the none option too?

File: chrome_extension/cardboard/manifest.json (hunk 2 above)
Comment (maximized) [reply chrome_extension/cardboard/manifest.json:19]: @rgossiaux, didn't you say you use 0.0.0.0? 

@akirabaruah I actually already included this in the most recent extension upload and it was accepted. However, maybe this is why it took longer.

https://developer.chrome.com/docs/extensions/develop/concepts/match-patterns has Chrome's suggestions. We can include just `http://localhost/*`, `http://127.0.0.1/*`, and `http://0.0.0.0/*`. Their documentation also suggests:

```
You can also use http://*:*/* to match localhost, IP addresses, and any port.
```

File: api/views.py (hunk 4 above)
Comment (maximized) [reply api/views.py:386]: Sure. I'll add a unit test.

File: hunts/src/AddPuzzleModal.js (hunk 3 above)
Comment (maximized) [reply hunts/src/AddPuzzleModal.js:65]: I'll just change both to "".

File: chrome_extension/cardboard/manifest.json (hunk 2 above)
Comment (akirabaruah) [reply chrome_extension/cardboard/manifest.json:19]: Ah thanks for the clarifications. SGTM.

I guess we want the `*`s because we want to run this extension on any website, i.e. arbitrary puzzle pages?

**Review by akirabaruah — APPROVED** [review]
LGTM with pending changes.

File: chrome_extension/cardboard/manifest.json (hunk 2 above)
Comment (maximized) [reply chrome_extension/cardboard/manifest.json:19]: No. I'm not authorizing using OAuth. I'm instead reading the csrftoken from the website's cookie. This is the list of URLs whose cookies I can read from. The asterisks are needed for folks various ports if you want to test it locally. 

The access to read any website's title and URL is from the permissions's activeTab

Comment (rgossiaux) [reply chrome_extension/cardboard/manifest.json:19]: > @rgossiaux, didn't you say you use 0.0.0.0?

Yeah I do, I was wondering why you changed to the `*:*/*` wildcard pattern since it looked like the only missing local development host was 127.0.0.1. I think `*:*/*` could match any random site, right? I don't think we want to ask people to give us permissions to read their cookies on the whole web if we can avoid it.

Comment (maximized) [reply chrome_extension/cardboard/manifest.json:19]: @rgossiaux I thought `*:*/*` would match anything with a colon but after testing apparently not. I changed it to "http://127.0.0.1/*"  in https://github.com/cardinalitypuzzles/cardboard/pull/737
